sx126x/spi_stm32wl.go

```diff
@@ -0,0 +1,106 @@
+//go:build stm32wlx
+// +build stm32wlx
+
+package sx126x
+
+import (
+	"device/stm32"
+	"errors"
+	"machine"
+)
+
+// New creates a new SX126x connection.
+func New(spi machine.SPI) *Device {
+	c := make(chan RadioEvent, 10)
+	d := Device{
+		spi:            spi,
+		radioEventChan: c,
+	}
+	// On STM32WL SoC, SPI3 is an internal bus to sx1262 (subghz)
+	// radio. It requires specific bus initialization.
+	if d.spi == machine.SPI3 {
+		d.SubGhzInit()
+	}
+	return &d
+}
+
+//SpiTx Receive byte over SPI
+func (d *Device) SpiTx(cmd uint8) uint8 {
+	r, _ := d.spi.Transfer(cmd)
+	return r
+}
+
+//SpiRx Receive byte over SPI
+func (d *Device) SpiRx() uint8 {
+	r, _ := d.spi.Transfer(0xFF)
+	return r
+}
+
+//SpiSetNss Sets the NSS line
+func (d *Device) SpiSetNss(state bool) {
+	if d.spi != machine.SPI3 {
+		for {
+			// Only STM32WL's internal sx126x radio is yet implemented
```

Comment:
Is going into an endless loop what you want here??

Reply:
You are right, this is not very "clean"....

But you fall here in a rare and specific use case: 

You want to use an external SPI SX126x radio on a STM32WL SoC (which already have a SX1262 module on the same die) -> Want to implement a multi-receiver, or a radio bridge ?

Ok, I agree. 
The best solution would be to implement this, but I'm not sure I'll have the time to do that in the next weeks. 

At the time I wrote the code, I identified three ways to address the problem: 

- Silent failure: Not very kind for who will try to run the driver in this setup
- Raising an error: I found it overkill to add error handling in such kind of basic functions.
- Endless loop: At least, it will stop the execution, and it'll be easy to track down the origin with a debugger.

How would you have dealt with this situation ? 

Thanks 

Olivier

Reply:
If this is something that can be detected at compile time then `panic` is probably the correct answer.

Reply:
panic is the good answer... 
It's working fine with it.